### DISCUSSION TITLE
Unable to run the example code for the C API

### DISCUSSION THREAD

**Discussion by Vaibtan** [Tencent/tquic#313] [General]
I was trying out the example given in [https://github.com/tquic-group/tquic-example-c/tree/main](url). After makefile compilation, I was getting this memory leak error when running the simple_client executable [https://github.com/tquic-group/tquic-example-c/blob/main/simple_client.c](url) . How should I resolve this? 
![img](https://github.com/Tencent/tquic/assets/93938070/95c87582-36a6-4502-a05d-2a66d0188633)

**Comment by glime-ai** [comment]
To resolve the memory leak error you encountered when running the `simple_client` executable from the TQUIC C example, you can follow these steps:

1. **Check for Updates**: Ensure that you are using the latest version of the TQUIC library and the example code. Sometimes, issues are fixed in newer versions.

2. **Review the Code**: Carefully review the `simple_client.c` code for any potential memory management issues. Ensure that all allocated memory is properly freed.

3. **Use Valgrind**: Utilize tools like Valgrind to identify the exact location of the memory leak. Valgrind can provide detailed information about memory usage and leaks.

4. **Consult Documentation**: Refer to the TQUIC documentation for any known issues or additional guidance on using the C API. The documentation can be found [here](https://tquic.net/docs/tutorial/c).

5. **Check GitHub Issues**: Look through the [GitHub issues](https://github.com/tencent/tquic/issues) for any similar problems reported by other users and potential solutions.

6. **Ask for Help**: If the issue persists, consider asking for help on the TQUIC GitHub discussions or issues page. Provide detailed information about the error and the steps you have taken to resolve it.

For more detailed guidance, you can refer to the following sources:
- [TQUIC C API Documentation](https://tquic.net/docs/tutorial/c)
- [TQUIC Example Code Repository](https://github.com/tquic-group/tquic-example-c)

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Comment by iyangsj** [comment]
@xiaofei0800

**Comment by iyangsj** [comment]
@Vaibtan The issue is caused by the `hashlru` crate and will be fixed in the next release. 

You can temporarily bypass it by modifying the CFLAGS in the [Makefile](https://github.com/tquic-group/tquic-example-c/blob/main/Makefile#L6):
```
CFLAGS = -I. -Wall -Werror -g -I$(TQUIC_DIR)/deps/boringssl/src/include/
```

**Reply by Vaibtan** [reply]
[tquic-server-output.txt](https://github.com/user-attachments/files/16261217/tquic-server-output.txt)
[tquic-client-output.txt](https://github.com/user-attachments/files/16261219/tquic-client-output.txt)
Okay so makefile compilation works now but there was no response by the server (as it should have sent "OK")

**Reply by iyangsj** [reply]
The log file `tquic-client-output.txt` shows that the client received a "HTTP/0.9 200 OK" message from the server.

<img width="1525" alt="image" src="https://github.com/user-attachments/assets/52ab5298-e5f2-459f-b942-1015e584808c">

**Reply by Vaibtan** [reply]
Oh sorry I missed it. Thanks for your help! 👍
